### PR TITLE
Add path filter based fuzzer jobs.

### PR DESCRIPTION
**Pull request by kgpai**
This PR adds path filtering support and thus enables us to have path based jobs for our continuous integration runs. In this particular PR along with path filtering support we add ability to have longer presto expression fuzzer runs , should any changes be made to velox/expression/, velox/core/ etc.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,647 +14,34 @@
...
-              echo "EOF"
-              false
-            fi
+          # Format is: <regex path-to-test> <parameter-to-set> <value-of-pipeline-parameter>
+          # Regex below will filter out paths with test in them.
+          mapping: |
+            velox/expression/((?!.*test).*).*  run-longer-expression-fuzzer true
+            velox/exec/((?!.*test).*).*  run-longer-expression-fuzzer true
+            velox/common/((?!.*test).*).*  run-longer-expression-fuzzer true
+            velox/core/((?!.*test).*).* run-longer-expression-fuzzer true
```
Comment (kagamiori) [inline .circleci/config.yml:41]: It would be good to also run longer for velox/vector/((?!.*test).*).*

File: .circleci/dist_compile.yml [hunk 2]
```diff
@@ -0,0 +1,697 @@
...
+    triggers:
+      - schedule:
+          cron: "0 3 * * *"
+          filters:
+            branches:
+              only:
+                - main
+    jobs:
+      - doc-gen-job
+      - linux-presto-fuzzer-run
```
Comment (kagamiori) [inline .circleci/dist_compile.yml:666]: Should nightly also run longer fuzzer test?

**Review by kagamiori — COMMENTED** [review]
LGTM overall. I just left two comments.

File: .circleci/config.yml (hunk 1 above)
Comment (kgpai) [reply .circleci/config.yml:41]: Sure I will add that, let me know if there are any other paths we might want to add for.

File: .circleci/dist_compile.yml (hunk 2 above)
Comment (kgpai) [reply .circleci/dist_compile.yml:666]: I think I will just change the nightlys to run for an hour.

**Review by assignUser — APPROVED** [review]
Neat, towards are more maintainable config and more efficient CI time! :grinning:

**Review by kagamiori — APPROVED** [review]
LGTM. Thank you for improving our fuzzer test to be more effective!

**Pull request closed without merging**